TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Infrastructure/Filters/CheckSupportTaskExistsFilter.cs

```diff
@@ -4,7 +4,7 @@
 
 namespace TeachingRecordSystem.SupportUi.Infrastructure.Filters;
 
-public class CheckSupportTaskExistsFilter(TrsDbContext dbContext, bool openOnly, SupportTaskType supportTaskType) : IAsyncResourceFilter
+public class CheckSupportTaskExistsFilter(TrsDbContext dbContext, bool openOnly, Func<SupportTaskType, bool> supportTaskTypePredicate) : IAsyncResourceFilter
```

Comment:
Might be simpler to pass a `params SupportTaskType[]` here instead of a delegate

Reply:
I did think about that tbh .. actually code looks a bit nicer so I've changed it!